### PR TITLE
Add MailChimp to staging.

### PR DESCRIPTION
**Pull request by budzanowski**
### Information

This is a part of #17434 PR that represents MailChimp integration into Calypso final step.
This is provided for review as a separate PR because the original (spike) PR is too big for the review to be comfortable. 

To test these elements in action please refer to #17434.

### CODE REVIEW

File: client/extensions/woocommerce/app/dashboard/index.js [hunk 1]
```diff
@@ -143,11 +151,22 @@ class Dashboard extends Component {
 			return <SetupTasksView onFinished={ this.onStoreSetupFinished } site={ selectedSite } />;
 		}
 
+		let manageView = null;
```
Comment (timmyc) [inline client/extensions/woocommerce/app/dashboard/index.js]: you could put `let manageView = <ManageOrdersView site={ selectedSite } />;` here and drop the `else` below.

File: client/extensions/woocommerce/app/dashboard/index.js [hunk 2]
```diff
@@ -80,6 +87,7 @@ class Dashboard extends Component {
 		if ( newSiteId && oldSiteId !== newSiteId ) {
 			this.props.fetchSetupChoices( newSiteId );
 			this.props.fetchOrders( newSiteId );
+			this.props.requestSyncStatus( newSiteId );
```
Comment (timmyc) [inline client/extensions/woocommerce/app/dashboard/index.js]: We should add this and `mailChimpConfigured` to `propTypes`

File: client/extensions/woocommerce/app/settings/email/mailchimp/getting-started.js [hunk 3]
```diff
@@ -38,25 +41,43 @@ const GettingStarted = localize( ( { translate, onClick } ) => {
 					<h3 className="mailchimp__getting-started-list-header">
 						{ translate( 'Connect with your customers through MailChimp' ) }
 					</h3>
-					<ul className="mailchimp__getting-started-list">
+					<ul className="mailchimp__getting-started-list" >
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/getting-started.js]: Could probably scratch this change and remove the space.

File: client/extensions/woocommerce/app/settings/email/mailchimp/getting-started.js [hunk 4]
```diff
@@ -38,25 +41,43 @@ const GettingStarted = localize( ( { translate, onClick } ) => {
...
 								<Gridicon icon="checkmark" size={ 18 } />
 								{ item }
 							</li>
 						) }
 					</ul>
-					<Button className="mailchimp__getting-started-button" onClick={ onClick }>
-						{ translate( 'Get started with MailChimp' ) }
-					</Button>
+					{ ! redirectToSettings && (
+						<Button className="mailchimp__getting-started-button" onClick={ onClick }>
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/getting-started.js]: Would it be cleaner to just have an onClick const that is assigned based upon `redirectToSettings`? Something like:

```
const mailChimpHref = redirectToSettings ? wizardLink : onClick;
...
<Button className="mailchimp__getting-started-button" onClick={ mailChimpHref }>
  { getStarted } // or not have getStarted as a const and just inline the translate string here
</Button>
```

File: client/extensions/woocommerce/app/settings/email/mailchimp/getting-started.js [hunk 5]
```diff
@@ -38,25 +41,43 @@ const GettingStarted = localize( ( { translate, onClick } ) => {
...
+					<p />
+				</Card> }
 		</div>
 	);
 } );
 
 GettingStarted.propTypes = {
 	onClick: PropTypes.func.isRequired,
+	isPlaceholder: PropTypes.bool,
+	redirectToSettings: PropTypes.bool,
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/getting-started.js]: we should add in `site` here too.

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js [hunk 6]
```diff
@@ -21,8 +21,12 @@ const NewsletterSettings = ( { storeData = {}, onChange, siteId, isRequesting, t
 	return (
 		<FormFieldset className="setup-steps__store-info-field">
 			<QueryMailChimpLists siteId={ siteId } />
-			<div>{ translate( 'Pick a list, you will not able to change it for now so pick carefully.' ) }</div>
-			<div>{ translate( 'Create your list at mailchimp.com if you have not done it aready.' ) }</div>
+			<div className="setup-steps__login-title">{ translate( 'Now choose a list to sync!' ) }</div>
+			<p>{ translate(
+					'Choose your list carefully as you wont be able to change it later. ' +
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: I'm not 💯 positive but thinking having these strings concatenated here might not be the proper way to do the translation call. Was this just to keep the link a certain length?

File: client/extensions/woocommerce/app/settings/email/mailchimp/getting-started.js (hunk 4 above)
Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/getting-started.js]: I don't think so. Once it is onClick and once it is href.

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js (hunk 6 above)
Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: Kelly did this bit - maybe she knows more. I think that it is OK. 
@kellychoffman

Comment (kellychoffman) [reply client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: Yeah, its to keep the line length shorter to keep the calypso bots happy. I believe its done in other places like this, but happy to change it if there's a better way to do it.

Comment (justinshreve) [reply client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: @timmyc @budzanowski @kellychoffman We asked in the Calypso channel awhile back and this is totally fine for translation calls.

Comment (timmyc) [reply client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: :TIL: thanks!

**Comment by alisterscott** [comment]
Is this still ready to merge or do the latest commits need review?

**Pull request closed without merging**